### ISSUE TITLE
Algorithm online practice

### ISSUE THREAD

**Issue by vcpudding** [fajieyuan/sigir2020_peterrec#6]
Hi Prof Yuan~ 

Do you have any suggestions on how to deploy the finetuned model for online applications for best efficiency and accuracy?

**Comment by fajieyuan** [comment]
Hi there,

PeterRec can be used to make recommendations for cold users who have few/no interactions but have rrich interactions from other domains. At Tencent, we have a lot of such scenarios. We can first pre-train a very large sequence model in domain A where there are a large number of users and most users have a large number of interactions, then we can save the  pretrained model and use it for domain B such as ads recommendation where most users have no interactions. The main advantage of PeterRec is that the pre-trained model can be used to serve many downstream tasks(cold start problem or user profile prediction), and could be realized in the parameter-efficent way. There is another scenario you can use PeterRec to predict user’s like or purchase behaviors since such behaviors are also very few. First Train the clicking sequence, and then transfer to purchase behaviors.


In our recent paper Conure(SIGIR2021), we attach some recommendation cases for new users, you can refer to https://github.com/fajieyuan/SIGIR2021-conure/tree/main/Case In general, Conure and PeterRec make verry similar recommendations by one-time transfer learning.

Best,

Fajie




2021年5月12日 下午5:10，Xinghua Zhu ***@***.******@***.***>> 写道：

 online applications for best efficiency and accuracy

**Comment by fajieyuan** [comment]
I suggest you use LambdaFM loss function for finetuning it is quite good.https://github.com/fajieyuan/SIGIR2020_peterrec/blob/master/PeterRec_cau_serial_lambdafm.py


2021年5月12日 下午5:49，原发杰 ***@***.******@***.***>> 写道：

Hi there,

PeterRec can be used to make recommendations for cold users who have few/no interactions but have rrich interactions from other domains. At Tencent, we have a lot of such scenarios. We can first pre-train a very large sequence model in domain A where there are a large number of users and most users have a large number of interactions, then we can save the  pretrained model and use it for domain B such as ads recommendation where most users have no interactions. The main advantage of PeterRec is that the pre-trained model can be used to serve many downstream tasks(cold start problem or user profile prediction), and could be realized in the parameter-efficent way. There is another scenario you can use PeterRec to predict user’s like or purchase behaviors since such behaviors are also very few. First Train the clicking sequence, and then transfer to purchase behaviors.


In our recent paper Conure(SIGIR2021), we attach some recommendation cases for new users, you can refer to https://github.com/fajieyuan/SIGIR2021-conure/tree/main/Case In general, Conure and PeterRec make verry similar recommendations by one-time transfer learning.

Best,

Fajie




2021年5月12日 下午5:10，Xinghua Zhu ***@***.******@***.***>> 写道：

 online applications for best efficiency and accuracy
